### ISSUE TITLE
Warning: Redis Pipelining commands on a Redis instance is deprecated and will be removed in Redis 5.0.0.

### ISSUE THREAD

**Issue by leoplct** [mperham/sidekiq#5202]
Ruby version: 3.0.2
Rails version: 6.1.4
Sidekiq / Pro / Enterprise version(s): 6.2.2 

Got this warning from Redis

```
Pipelining commands on a Redis instance is deprecated and will be removed in Redis 5.0.0.

redis.multi do
  redis.get("key")
end

should be replaced by

redis.multi do |pipeline|
  pipeline.get("key")
end

(called from .../bundle/ruby/3.0.0/gems/sidekiq-6.2.2/lib/sidekiq/client.rb:189:in `block in raw_push'}

```

**Comment by adamniedzielski** [comment]
This is fixed in the newer versions of Sidekiq.

**Comment by slhck** [comment]
FWIW, I have to fix a legacy system that uses Sidekiq 5.2 and an old Redis version, and both cannot be upgraded for various reasons.

Explicitly requiring an older version of the `redis` gem makes the warning disappear:

```
gem 'redis', '< 4.6'
```

Run `bundle update` and you should see that that Gem regressed to 4.5.x. This works, but of course you should not upgrade Redis either.

:warning: I **do not recommend this** to anyone running a system where you could simply upgrade Sidekiq.

**Comment by kristenhazard** [comment]
Thank you for this thread. I, too have a legacy app (Rails 4.2) that was constrained in the sidekiq version I could update to. This seems to have removed the logging for me as well.
